source/RialtoGStreamerMSEAudioSink.cpp

```diff
@@ -76,10 +79,19 @@ static GstStateChangeReturn rialto_mse_audio_sink_change_state(GstElement *eleme
         }
         if (priv->isVolumeQueued)
         {
-            client->setVolume(priv->volume);
+            client->setVolume(priv->targetVolume, kDefaultVolumeDuration, kDefaultEaseType);
             priv->isVolumeQueued = false;
         }
-
+        if (priv->isAudioFadeQueued)
+        {
+            AudioFadeConfig audioFadeConfig;
+            {
+                std::lock_guard<std::mutex> lock(priv->audioFadeConfigMutex);
+                audioFadeConfig = priv->audioFadeConfig;
+            }
+            client->setVolume(priv->targetVolume, audioFadeConfig.duration, audioFadeConfig.easeType);
```

Comment:
targetVolume isnt set when we set the fade volume,, I think it was correct how it was before audioFadeConfig.volume